src/o365/graph/commands/subscriptions/subscription-add.ts

```diff
@@ -0,0 +1,244 @@
+  resource: string;
+  changeType: string;
+  notificationUrl: string;
+  expirationDateTime: string;
+  clientState: string;
+}
+
+export const DEFAULT_EXPIRATION_DELAY_IN_MINUTES_PER_RESOURCE_TYPE = {
+  // User, group, other directory resources	4230 minutes (under 3 days)
+  "users": 4230,
+  "groups": 4230,
+  // Mail	4230 minutes (under 3 days)
+  "/messages": 4230,
+  // Calendar	4230 minutes (under 3 days)
+  "/events": 4230,
+  // Contacts	4230 minutes (under 3 days)
+  "/contacts": 4230,
+  // Group conversations	4230 minutes (under 3 days)
+  "/conversations": 4230,
+  // Drive root items	4230 minutes (under 3 days)
+  "/drive/root": 4230,
+  // Security alerts	43200 minutes (under 30 days)
+  "security/alerts": 43200
+};
+export const DEFAULT_EXPIRATION_DELAY_IN_MINUTES = 4230;
+export const SAFE_MINUTES_DELTA = 1;
+
+class GraphSubscriptionAdd extends GraphCommand {
+  public get name(): string {
+    return `${commands.SUBSCRIPTION_ADD}`;
+  }
+
+  public get description(): string {
+    return 'Creates a Microsoft Graph subscription';
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    if (this.verbose) {
+      cmd.log(`Adding subscription to resource '${args.options.resource}' and change type '${args.options.changeType}'...`);
+    }
+    if (this.debug) {
+      cmd.log(`Notification will be sent to ${args.options.notificationUrl}.`);
+    }
+
+    const body: any = {
+      changeType: args.options.changeType,
+      resource: args.options.resource,
+      notificationUrl: args.options.notificationUrl,
+      expirationDateTime: this._getExpirationDateTimeOrDefault(cmd, args)
+    };
+
+    if (args.options.clientState) {
+      body["clientState"] = args.options.clientState;
+    }
+
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/subscriptions`,
+      headers: {
+        accept: 'application/json;odata.metadata=none',
+        'content-type': 'application/json'
+      },
+      body,
+      json: true
+    };
+
+    request
+      .post(requestOptions)
+      .then((res: any): void => {
+        cmd.log(res);
+
+        if (this.verbose) {
+          cmd.log(vorpal.chalk.green('DONE'));
+        }
+
+        cb();
+      }, (err: any) => this.handleRejectedODataJsonPromise(err, cmd, cb));
+  }
+
+  private _getExpirationDateTimeOrDefault(cmd: CommandInstance, args: CommandArgs): string {
+    if (args.options.expirationDateTime) {
+      if (this.verbose) {
+        cmd.log(`Expiration date time is specified (${args.options.expirationDateTime}).`);
+      }
+      return args.options.expirationDateTime;
+    }
+
+    if (this.verbose) {
+      cmd.log(`Expiration date time is not specified. Will try to get appropriate maximum value`);
+    }
+
+    const perResource: any = DEFAULT_EXPIRATION_DELAY_IN_MINUTES_PER_RESOURCE_TYPE;
+    for (let resource in perResource) {
+      if (args.options.resource.indexOf(resource) > -1) {
```

Comment:
By changing this if condition to:

```ts
if (args.options.resource.indexOf(resource) < 0) {
  continue;
}
```

we can avoiding indenting the whole body increasing readability